docs/parameters.md

```diff
@@ -0,0 +1,470 @@
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+        type: json
+```
+
+#### Passing the entire payload at once
+
+If the user has the entire parameters payload prepared to be sent, there are
+several options of passing it:
+- Inline embedded raw JSON (applicable only for non-sensitive data)
+```yaml
+  ...
+  parametersFrom:
+    - value:
+        simple: text
+        obj:
+          child: true
+        list:
+        - item1
+        - item2
+```
+which will be transformed into the following payload to be sent to the broker:
+```json
+{
+  "simple": "text",
+  "obj": {
+    "child": true
+  },
+  "list": ["item1", "item2"]
+}
+```
+- Secret key with JSON payload:
+```yaml
+  ...
+  parametersFrom:
+    - secretKeyRef:
+        name: mysecret
+        key: mykey
+```
+- ConfigMap key with JSON payload
+```yaml
+  ...
+  parametersFrom:
+    - configMapKeyRef:
+        name: myconfig
+        key: mykey
+```
+
+### Conflict resolution
+
+Within the scope of a `parameters` or a `parametersFrom` block, if there are 
+duplicate top-level names then the last one in each section will take precedence.
+If there are duplicate top-level names across `parameters` and `parametersFrom` 
+blocks then the `parameters` always takes precedence over `parametersFrom`.
+
+This conflict resolution strategy is consistent with the one supported by Kubernetes
+for environment variables (`env` and `envFrom` fields) in the `Pod` resource.
+
+For example, for the structure like this
+```yaml
+  ...
+  parameters:
+    - name: param
+      value: 1
+    - name: param
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key
+    - name: param
+      value: 2
+  parametersFrom:
+    - name: param
+      secretRef:
+        name: mysecret
+```
+the parameters JSON structure to be sent to the broker is going to be
+```json
+{
+  "param": "2"
+}
+```
+
+To avoid conflicts for `parametersFrom` (for example, in the case when multiple 
+secrets of the same structure need to be passed to the broker), the user can specify 
+a `name` field which will produce an extra nesting level.
+
+For example, let's say we have the following structure in a `Secret`:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: MWYyZDFlMmU2N2Rm
+```
+
+To inject credentials from two instances of such a `Secret`, we can avoid the 
+naming conflict by providing unique names:
+```yaml
+  ...
+  parametersFrom:
+    - name: master
+      secretRef:
+        name: secret1
+    - name: slave
+      secretRef:
+        name: secret2
+```
+which will result in the following JSON parameters object:
+```json
+{
+  "master": {
+    "username": "root",
+    "password": "letmein"
+  },
+  "slave": {
+    "username": "foo",
+    "password": "bar"
+  }
+}
+
+```
+
+## Advanced example
```

Comment:
@duglin @pmorie please take a look at the fully-featured example.
I'm not sure if it is useful actually, covering all the cases in single example is overwhelming and not possible.
I tried to cover most common cases though.